### DISCUSSION TITLE
Access ioctl result?

### DISCUSSION THREAD

**Discussion by daandemeyer** [bpftrace/bpftrace#2052] [Q&A] [answered]
I want to intercept ioctl syscalls at exit and inspect their results after they finish. Unfortunately, only the return value and syscall number are accessible in `tracepoint:syscalls:sys_exit_ioctl`. Is there a way to access the arguments that were passed to ioctl() (like in `tracepoint:syscalls:sys_enter_ioctl`)?

**Comment by fbs** [comment, marked as answer]
Syscall tracepoints don't save arguments, they just read registers which probably are modified when the function exists. What you have to do is save arguments yourself.

Opensnoop.bt does the same:

```
tracepoint:syscalls:sys_enter_open,
tracepoint:syscalls:sys_enter_openat
{
	@filename[tid] = args->filename;
}

tracepoint:syscalls:sys_exit_open,
tracepoint:syscalls:sys_exit_openat
/@filename[tid]/
{
	$ret = args->ret;
	$fd = $ret > 0 ? $ret : -1;
	$errno = $ret > 0 ? 0 : - $ret;

	printf("%-6d %-16s %4d %3d %s\n", pid, comm, $fd, $errno,
	    str(@filename[tid]));
	delete(@filename[tid]);
}
```
